### ISSUE TITLE
Builder

### ISSUE THREAD

**Issue by kazu-yamamoto** [yesodweb/wai#675]
I guess that it's time to switch `Builder` of `blaze-builder` to that of `bytestring`.
We need to research the impact of this change.

**Comment by mgmeier** [comment]
After conferring with @snoyberg via mail, that change looks rather painless since
1. `blaze-builder` re-exports `bytestring`'s `Builder` type these days
2. `blaze-bilder`'s chunked (HTTP) encoding isn't used, just API calls that `bytestring` provides as well.

Dropping `blaze-builder` should therefore be transparent to consumers of all the packages in this repo. Since time is an issue here (`blaze-builder` still fails to merge a PR that is in line with the Semigroup/Monoid-Proposal, and therefore doesn't compile with GHC 8.4.1 at all), I'd take a stab at this issue and send a PR when done.

**Comment by mgmeier** [comment]
a heads up: I had to open a PR here kazu-yamamoto/logger#159 to remove the (unused) `blaze-builder` dependency from `wai-logger`. So without a published new version of that package, these here depending on it won't compile.

**Comment by kazu-yamamoto** [comment]
@mgmeier I completely agree with you.
Some days ago, I tried to remove `blaze-builder` for all packages relating to `wai`. But it appeared that a lot of things should done and I gave up.

**Comment by mgmeier** [comment]
@kazu-yamamoto could you reopen this ticket for discussion? There are still some cases where I need input on how to proceed (esp. `warp` package). Or do you prefer to discuss via e-mail?

**Comment by mgmeier** [comment]
FYI: For wai-websockets, I'm waiting on jaspervdj/websockets#170 ("Remove dependency on `blaze-builder`") to merged and released to depend on the latest `websockets` package. I'll create another PR for that.

**Comment by kazu-yamamoto** [comment]
I might misunderstand your intention. Sorry.
I would love to continue the discussion here.

**Comment by mgmeier** [comment]
@sjakobi and me did everything in the same PR after all, it's been merged, so I think this issue can be closed. Thank you @kazu-yamamoto for your time and especially for your quick responses.

**Comment by kazu-yamamoto** [comment]
Thanks!
